OpenSim/Common/Component.cpp

```diff
@@ -1098,19 +1099,27 @@ void Component::markPropertiesAsSubcomponents()
 
 // mark a Component as a subcomponent of this one. If already a
 // subcomponent, it is not added to the list again.
-void Component::markAsPropertySubcomponent(Component* component)
+void Component::markAsPropertySubcomponent(const Component* component)
 {
     // Only add if the component is not already a part of this Component
     // So, add if empty
-    SimTK::ReferencePtr<Component> compRef(component);
+    SimTK::ReferencePtr<Component> compRef(const_cast<Component*>(component));
     if (_propertySubcomponents.empty() ){
-        _propertySubcomponents.push_back(SimTK::ReferencePtr<Component>(component));
+        // Must reconstruct the reference pointer in place in order
```

Comment:
@tkuchida You are right it isn't necessary. `find` should return `end()` even if empty.
@stavness it would be worth simplifying and also merging with latest changes on master. I am hoping that will also fix the issue with the choco install of swig on AppVeyor.